stickyuploads/tests/test_views.py

```diff
@@ -33,13 +33,16 @@ def test_upload(self):
             response = self.client.post(self.url, data)
         self.assertEqual(response.status_code, 200)
         result = json.loads(response.content.decode('utf-8'))
-        storage = TempFileSystemStorage()
         filename = os.path.basename(self.temp_name)
+        # We can't test full filename value because there are random letters
+        self.assertIn('filename', result)
+        self.assertIn(filename, result['filename'])
+        self.assertIn('stored', result)
         expected = {
             'is_valid': True,
-            'filename': filename,
+            'filename': result['filename'],
```

Comment:
I use mocking for last resort because I don't want to hide functionality unless have to.
I know that checking for something by using the same thing is not good but in this case I want to test that:

1. We have `filename` in `result`
2. We have `stored` in `result`
3. File name of the file is used in returned `filename`
4. The whole result is a dict with only 'is_valid', 'filename', 'url' and 'stored' keys
5. The value of 'is_valid' is True,
6. The value of `url` is None

The only way to test for 4 is to construct dict with only these keys and to compare with it. The problem is that I don't know the values of some of the keys.
I can't mock filename creation because I want to test the the original file name is in the `filename`